### ISSUE TITLE
Process packages parallelly

### ISSUE THREAD

**Issue by marxin** [rpm-software-management/rpmlint#502]
The following loop can be processed in a parallel way:
https://github.com/rpm-software-management/rpmlint/blob/3fb9d6ce2a1538103aba946931db0adc147eb4a5/rpmlint/lint.py#L146-L148

**Comment by scarabeusiv** [comment]
It can't until the pkg.py is redone as it stores some of the data in the same location unfortunately.

**Comment by marxin** [comment]
I see, please assign it to me then.

**Comment by marxin** [comment]
> It can't until the pkg.py is redone as it stores some of the data in the same location unfortunately.

No, apparently it's not the problem. Each `rpm` is extra to its own directory. The problem seems to be a shared resource:
`self.checks` that likely have an internal state..

**Comment by marxin** [comment]
And the crucial shared resource is `rpm` (RPM binding).
Apart from that also `_magic` is not thread safe.

**Comment by marxin** [comment]
please assign it to me

**Comment by marxin** [comment]
One another option is to run parallelly all checks for each package. Let me try it...
